### ISSUE TITLE
Tuner tutorial rebuilds model rather than saving it

### ISSUE THREAD

**Issue by dansbecker** [keras-team/keras#15789]
## URL(s) with the issue:
https://www.tensorflow.org/tutorials/keras/keras_tuner

## Description of issue (what needs changing):
The bottom of this tutorial says `Re-instantiate the hypermodel and train it with the optimal number of epochs from above.` But there's no reason to train the model twice when you could use the callback to save the model at the best epoch. I and others have speculated [here](https://twitter.com/HamelHusain/status/1470492047404580865) that the call to re-train the model is supposed to be on the full data (with no validation split). If that's the intention, the resulting change would be to remove `validation_split=0.2` from the final `fit` call.

Otherwise, it seems you should add a `ModelCheckpoint` callback to the save the best model when you first fit the model... and then not refit the model.

**Comment by haifeng-jin** [comment]
Thanks for the issue. Seems the docs needs to be updated.

**Comment by gowthamkpr** [comment]
@haifeng-jin @dansbecker I have modified the example and save the model weights with best val accuracy instead of retraining it. Please take a look at the [gist](https://colab.research.google.com/gist/gowthamkpr/09c09341aed90dfa4219ecf58d6488ca/keras_tuner.ipynb) and let me know. I can create a PR. Thanks!

**Comment by haifeng-jin** [comment]
I mean adding a few lines to the tutorial to explain why you may prefer to retrain the model should be good enough. Some users would search with fewer numbers of epochs just to get some quick results.

**Comment by haifeng-jin** [comment]
Use https://github.com/keras-team/keras-tuner/issues/687 to track. So closing this one.
